python-client/giskard/models/shap_result.py

```diff
@@ -0,0 +1,124 @@
+from enum import Enum
+from typing import Any, Iterable
+from dataclasses import dataclass
+
+import wandb
```

Comment:
The point is that I am going to use "shap_results.py" module just in case, when I need to import ShapResults class, which requires wandb. So I do not need anything else from this module, which do not use wandb. i.e. I am gonna use this module only if wandb is installed. Otherwise, I need to put wandb import to all private functions and ShapResults.to_wandb() which looks like an overhead. WDYT?

Reply:
since we decided to make `explain_with_shap` a global function that returns `ShapResult`, so that users can use it without necessarily wanting to log it with wandb (maybe later we want to log it to mlflow or other mlops).

several imports of the same libraries is not an issue in python btw, see: https://stackoverflow.com/questions/37067414/python-import-multiple-times. 

If you don't like the idea of import `wandb` in every plotting function, you can refactor them into `wandb_utils.py`, I think that's a cleaner solution.

Reply:
But if I put plotting functions into `wandb_utils.py`, I will also get an error, but defined by us. Do you expect such behaviour and we are ok with it?

Reply:
`try:
    import wandb  # noqa
    from wandb.wandb_run import Run
except ImportError as e:
    raise ImportError("The 'wandb' python package is not installed. To get it, run 'pip install wandb'.") from e`